CONTRIBUTING.md

```diff
@@ -1,9 +1,75 @@
-## Code Checks Before Committing
+# Contribution Guidelines
 
-We use Husky and lint-staged to check your staged files.
+This project is open to contributors of all levels. While it is primarily geared toward members of the Code Café Discord server,
+anyone with an interest in contributing to open source can pick up create a bug request, feature request, or open a PR that addresses an outstanding issue.
 
-When you commit, Husky triggers lint-staged, which runs tools like ESLint and Prettier on changed files.
+We ask that if you wish to contribute, you follow the best practices listed below.
 
-If any issues are found, the commit is blocked until they’re fixed. Check your terminal output for linting or formatting errors.
+## For All Contributors
+
+General guidance for contributors. For maintainer-specific info, see the [For Maintainters](#for-maintainers) section.
+
+### Do :heavy_check_mark:
+
+- Read through existing issues before opening a new bug or feature request to avoid creating duplicate issues.
+- Leave a comment on an open issue if you would like to work on it. If the issue is available to work on, a maintainer will assign the issue to you as soon as possible.
+- Keep PRs scoped to a single issue.
+  - Always prefer multiple small PRs over one larger PR that addresses multiple things. This helps better prevent "scope creep", and makes it easier for maintainers and contributors alike. It also allows more contributors to work on issues in parallel.
+- Ask for help if you have a question.
+  - Feel free to leave comments on the issue if you have any questions around implementation or if you need help of any kind.
+- Fork the repo and create a new branch on your fork in order to start work on your assigned issue.
+  - If you become a regular contributor, you may request to be added as a collaborator on the project so that you can work from a direct clone rather than a fork.
+- Name your feature branch something descriptive of your task. Including the issue number is common practice.
+  - For example, if you are working on issue number 42 and performing a refactor on the navbar component, a good branch name would be `42-refactor-navbar-component`.
+- Ensure that your code works, tests pass and your code is properly linted and formatted prior to opening a PR.
```

Comment:
Might be worth adding that, once a PR is created and the user is happy with it, to formally request review from @\CodeCafeCommunity/website?

Reply:
Or create a discord channel specifically for contributors. And assign a role only when someone has actively taken on an issue. I don't know if that can be done with an automatic process.

Reply:
Ooh neat, I think we'd probably have to do it manually, but I agree with having a channel.

Reply:
I definitely like the idea of having a dedicated channel and Contributor role in the discord server! I think maybe they should receive the role after having a PR merged in. Once we have that set up we can add a section about it to this document